src/Authentication/Authentication/Cmdlets/ConnectMgGraph.cs

```diff
@@ -196,20 +209,46 @@ private async Task ProcessRecordAsync()
                         break;
                 }
 
-                try
+                
+                var (context, authError) = await Authenticator.AuthenticateAsync(authContext, ForceRefresh, _cancellationTokenSource.Token,
+                   type =>
+                   {
+                       // Early Detection of FallBack to DeviceCode
+                       if (type == AuthProviderType.DeviceCodeProviderFallBack)
+                       {
+                           WriteWarning(Resources.DeviceCodeFallback);
+                       }
+                   });
+
+                switch (authError.AuthErrorType)
                 {
-                    // Save auth context to session state.
-                    GraphSession.Instance.AuthContext = await Authenticator.AuthenticateAsync(authContext, ForceRefresh, _cancellationTokenSource.Token);
-                }
-                catch(Exception ex)
-                {
-                    throw ex;
+                    case AuthErrorType.None:
+                    case AuthErrorType.FallBack:
+                        {
+                            GraphSession.Instance.AuthContext = context;
+                            //Fallback was due to an Exception, ie Browser Could not be Opened.
+                            if (authError.AuthErrorType == AuthErrorType.FallBack)
+                            {
+                                WriteWarning(Resources.DeviceCodeFallback);
+                            }
+                            WriteObject("Welcome To Microsoft Graph!");
+                            break;
+                        }
```

Comment:
Isn't this a duplicate of line [219](https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/618/files#diff-6765c82282b378bf2f4edc2b785424abe653b006678a02f6b9f4cc0317036bc0R219)?